docs/source/newsfragments/4542.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fix bug where :meth:`cancelling <cocotb.task.Task.cancel>` a Task awaiting on a :class:`.First` or :class:`.Combine` doesn't cancel the underlying waiters leading to reduction in performance or concealling test failures.
```

Comment:
```suggestion
Fix a bug where :meth:`cancelling <cocotb.task.Task.cancel>` a Task awaiting on a :class:`.First` or :class:`.Combine` doesn't cancel the underlying waiters leading to reduction in performance or concealing test failures.
```

Reply:
Resolved but not applied, was this intentional?

Reply:
Yeah I had fixed it on one computer, marked it resolved, but forgot to push, switched to another computer and pushed again without the changes. One more time...